docs/modules/ROOT/pages/BasicConcept.adoc

```diff
@@ -36,56 +53,180 @@ filter4 = "Power Nice" -- nick filter maches noone
+
+However, you probably don't feel like you're any closer to using the system. That's a fair feeling right now, especially if you're a newbie.
+
+In order to make things more clear, let's work through an example. In this case, our goal is to write a script that tells us the current recipe on a Constructor.
+
+_Note: The steps in this example are provided with the intent to get you up-and-running as quickly as possible. To better understand each of these steps, see the xref::lua/guide/index.adoc[Lua Guide] for more information._
+
+==== Step 1: Finding the Component
+
+The first step in your code is to find the Constructor on the network and save a *reference* to it in a variable. By saving a reference, we can then refer to its respective functions and properties very easily.
+
+This is accomplished by using the `component.proxy` function. We can use the function by itself and feed it the GUID of the Constructor in order to save a reference to it:
+
+`constructor = component.proxy('0123456789...')`
+
+But we used the Network Manager to name our Constructor "WireFactory Constructor1", and we'd like to use that nick instead of a long string of random numbers when we search for our component. Fortunately, we can let `component.findComponent` find the GUID for us using our nicks.
+
+`constructor = component.proxy(component.findComponent(
+    "WireFactory Constructor1")[1])`
+
+Is it pretty? No. Is it functional? Yes. 
+
+The function returns an array containing GUIDs of every component in the network whose nick matched the query string. Our query string should only return our one Constructor, so now we just need to reference the first element in the array. We add `[1]` at the end of our `findComponent` function call in order to reference the first item it returns. 
+
+_Note: in Lua, arrays start at one, which is unlike most programming languages._
+
+==== Step 2: Using the 'getRecipe' Function
+
+Now that we have a reference to our Constructor saved in the variable `constructor`, we can reference its functions and properties with very little effort.
+
+A look at the description for the https://docs.ficsit.app/ficsit-networks/latest/Reflection.html#Class-Manufacturer[Manufacturer component] informs us that it is actually the base class for _all_ machines that use a recipe, including our Constructor. We don't have to scroll much farther to see that `getRecipe` is the function in that class that we need to achieve our goal!
+
+Looking at the description of `getRecipe`, we see the following:
+
+"Get Recipe `getRecipe` (Class(Recipe) Recipe `recipe` _out_)"
+
+This might look like gibberish at first, but here's a look at each of the pieces individually:
+
+- *Get Recipe* - More readable, "localized name" of the function
+- `getRecipe` - Name of the function within the Reflection System
+- *()* - Parentheses indicate a list of the inputs and outputs of the function (known as "arguments") separated by commas
```

Comment:
We prefer parameters, as in unreal engine you refer to them as "input parameters" and "output parameters"...
arguments is most commonly used for input paramaters exclusively